pkg/generator/migration/generator.go

```diff
@@ -0,0 +1,104 @@
+package migration
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+	plugin "github.com/golang/protobuf/protoc-gen-go/plugin"
+	"github.com/gophersbd/ormpb/pkg/descriptor"
+	"github.com/gophersbd/ormpb/pkg/dialect"
+	"github.com/gophersbd/ormpb/pkg/generator/common"
+	"github.com/gophersbd/ormpb/pkg/validation"
+)
+
+type generator struct {
+	reg *descriptor.Registry
+}
+
+// NewGenerator return Generator interface
+func NewGenerator(reg *descriptor.Registry) common.Generator {
```

Comment:
you can return *generator to avoid import cycle

Reply:
`pkg/generator/migration/generator.go:22:45:warning: exported func NewGenerator returns unexported type *migration.generator, which can be annoying to use (golint)`

Reply:
why not make it public

Reply:
Done